modules/01 JavaScript Programming/Assignments/19/answer19

```diff
@@ -0,0 +1,100 @@
+
+
+        function validateCreditCard(x){
+            var arrNumbers = x.split("");
+            var flag = true;
+            var validationStatus ="NOT VALEID : ";
+            var total=0;
+            if (arrNumbers.length !== 16){
+                validationStatus+=" --you have to fill 16 numbers";
+                flag =false;
+
+            }
+
+
+            var bool=arrNumbers.some(function(item){
+                return(( typeof +item !== 'NAN') &&  (arrNumbers.some(function(item2){
+
+                   return ( typeof +item2 !== 'NAN' && +item2!==+item);
+
+                }))) ;   
+            });
+
+                if (!bool){
+                    validationStatus = " --you must have 2 different numbers";
+                    flag=false;
+
+                }
+
+
+
+
+
+
+            
+           var isNumbers= arrNumbers.every(function(item) {
+              return !isNaN(+item);
+            });
+
+            if(!isNumbers){
+                validationStatus+=" --you have to enter just numbers without chars";
+                flag =false;
+
+
+            }
+
+            if(!(+arrNumbers[arrNumbers.length-1]%2===0)){
+               validationStatus+=" --the last digit must be even ";
+               flag =false;
+
+
+
+            }
+
+                for(var i in arrNumbers) {
+                    
+                    total += +arrNumbers[i]; 
+                
+                }
+
+                if(total<=16){
+                    validationStatus+=" --you must have sum of numbers that is greater than 16  ";
+
+                    flag =false;
+                  
+                }
+
+(flag===true) ? console.log("you have a valid visa card number") : console.log( validationStatus )
+
+
+
+
+        }
+
+        // Standard Question
+
+        var creditNumber1="6666666666661666";
+        console.log(creditNumber1);
+        validateCreditCard(creditNumber1);
+
+
+
+        // -----------------------  Bonus 1 we must add this code 
+        function bonus(creditNumber2){
+        var arr;
+        if(creditNumber2.indexOf("-")>=0){
+         arr = creditNumber2.replace("-","");
+         arr =  arr.replace("-","");
+         arr =  arr.replace("-","");
+
+        }
+        validateCreditCard(arr);
+
+    }
+
+
+    
+         var creditNumber2="9999-7777-8888-0007";
+         console.log(creditNumber2);
+         bonus(creditNumber2);
+
```

Comment:
Correct. Please format your code using "Visual Code" before submitting in order to format line breaks and spaces